### DISCUSSION TITLE
SMTP sending mail error

### DISCUSSION THREAD

**Discussion by Mickyjohn** [dani-garcia/vaultwarden#2424] [Q&A]
I am self hosting the Vaultwarden on Almalinux 8.5. The postfix works by using echo command to send mail from terminal
### Subject of the issue
SMTP mail error: Error sending SMTP test email
SMTP Client error: internal client error: No compatible authentication mechanism was found

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.24.0
* Web-vault version: v2.25.1
* Running within Docker: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.35.4
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** SMTP_HOST, SMTP_SSL, SMTP_PORT, SMTP_FROM

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": false,
  "_ip_header_enabled": true,
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*********.*****************.***",
  "domain_origin": "*****://*********.*****************.***",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": "*********",
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_org_name": "Wenzel Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": false,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": true,
  "smtp_accept_invalid_hostnames": true,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": false,
  "smtp_from": "***********@*****************.***",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "*********.*****************.***",
  "smtp_password": "***",
  "smtp_port": 25,
  "smtp_ssl": false,
  "smtp_timeout": 150,
  "smtp_username": "",
  "templates_folder": "data/templates",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

* Install method:  Docker Image

* Clients used: 
* Web vault

* Other relevant details:

### Steps to reproduce
docker run -d --name vaultwarden \
  -e SMTP_HOST=bitwarden.mydomain.com \
  -e SMTP_FROM=bitwarden@mydomain.com \
  -e SMTP_PORT=25 \
  -e SMTP_SSL=false \
  -e SMTP_USERNAME= \
  -e SMTP_PASSWORD= \
  -v /vw-data/:/data/ \
  -p 8088:80 \
  vaultwarden/server:latest

### Expected behaviour
The mail should be sent

### Actual behaviour
and in Postfix mail logs : -
Apr 19 07:48:19 bitwarden postfix/smtpd[188087]: connect from unknown[172.17.0.2]
Apr 19 07:48:19 bitwarden postfix/smtpd[188087]: discarding EHLO keywords: CHUNKING
Apr 19 07:48:19 bitwarden postfix/smtpd[188087]: lost connection after EHLO from unknown[172.17.0.2]
Apr 19 07:48:19 bitwarden postfix/smtpd[188087]: disconnect from unknown[172.17.0.2] ehlo=1 commands=1

### Troubleshooting data
Postfix configuration:

postconf -n

alias_database = hash:/etc/aliases
alias_maps = hash:/etc/aliases
command_directory = /usr/sbin
compatibility_level = 2
daemon_directory = /usr/libexec/postfix
data_directory = /var/lib/postfix
debug_peer_level = 2
debugger_command = PATH=/bin:/usr/bin:/usr/local/bin:/usr/X11R6/bin ddd $daemon_directory/$process_name $process_id & sleep 5
html_directory = no
inet_interfaces = all
inet_protocols = all
mail_owner = postfix
mailq_path = /usr/bin/mailq.postfix
manpage_directory = /usr/share/man
meta_directory = /etc/postfix
mydestination = localhost.localdomain, localdomain, localhost, localhost.localdomain, localhost
myhostname = [bitwarden.mydomain.com](http://bitwarden.mydomain.com/)
mynetworks = 127.0.0.0/8, localhost, 172.17.0.0/16, 172.17.0.2
newaliases_path = /usr/bin/newaliases.postfix
queue_directory = /var/spool/postfix
readme_directory = /usr/share/doc/postfix/README_FILES
relayhost =
sample_directory = /usr/share/doc/postfix/samples
sendmail_path = /usr/sbin/sendmail.postfix
setgid_group = postdrop
shlib_directory = /usr/lib64/postfix
smtp_tls_CApath = /etc/pki/tls/certs
smtp_tls_security_level = may
smtpd_tls_cert_file = /etc/pki/tls/certs/postfix.pem
smtpd_tls_key_file = /etc/pki/tls/private/postfix.key
smtpd_tls_security_level = may
unknown_local_recipient_reject_code = 550

**Comment by BlackDex** [comment]
I'm not sure how you use echo to test, i normally use telnet or nc for these tasks. But besides that, i think postfix only allows login on port 587 and not 25 which is only used to receive emails. And it could also be that it only allows login when you have ssl or tls enabled.

I would first check what happens if you use port 587 with ssl or tls enabled.

**Comment by Mickyjohn** [comment]
If I use Port 587 and Force TLS enable it gives and error: Error sending SMTP test email
SMTP Connection error: Connection refused (os error 111) 
which means it even dont connect to Postfix mail server as there are not logs in mail server.
Although:
# firewall-cmd --list-all
public (active)
  target: default
  icmp-block-inversion: no
  interfaces: ens3
  sources: 
  services: cockpit dhcpv6-client http https ssh
  ports: 80/tcp 443/tcp 25/tcp 587/tcp
  protocols: 
  forward: no
  masquerade: yes
  forward-ports: 
  source-ports: 
  icmp-blocks: 
  rich rules
 
For a while i am not interested in SSL or login credentials to send mail because i am self hosting.
Port 25 do not use SSL thats why i wish to use it.
Echo mail test i use it as follows:
$ echo "test mail" | sendmail myname@mydomain.com
mail sent sucessfully as can be seen from mail log
Apr 19 10:15:15 bitwarden postfix/pickup[198315]: 81A8D41ABD63: uid=0 from=<root>
Apr 19 10:15:15 bitwarden postfix/cleanup[216630]: 81A8D41ABD63: message-id=<20220419081515.81A8D41ABD63@bitwarden.mydomain.com>
Apr 19 10:15:15 bitwarden postfix/qmgr[198316]: 81A8D41ABD63: from=<root@bitwarden.mydomain.com>, size=314, nrcpt=1 (queue active)
Apr 19 10:15:15 bitwarden postfix/smtp[216632]: 81A8D41ABD63: to=<myname@mydomain.de>, relay=mail.mydomain.com[10.59.2.173]:25, delay=0.09, delays=0.03/0.02/0.02/0.02, dsn=2.0.0, status=sent (250 2.0.0 Ok: queued as 912235833233)
Apr 19 10:15:15 bitwarden postfix/qmgr[198316]: 81A8D41ABD63: removed

can i send mail without login credential and SSL ?

**Reply by BlackDex** [reply]
I would also checkout https://github.com/dani-garcia/vaultwarden/wiki/SMTP-Configuration btw.

Regarding not using SSL and using port 25, that all depends on the configuration of Postfix, for which this place is not the correct place to discuss Postfix configurations.

In theory it should be possible to allow specific clients to relay emails without the need to login and use port 25. But how that works in your setup i have no clue, and i suggest to ask these kind of questions at a Postfix forum or something.

On the vaultwarden side it definitely is possible, the error you get is that the mail-library isn't receiving valid Login response from the mail-server, and thus doesn't know how to continue.

**Reply by Mickyjohn** [reply]
I further debug the mail log and think it is not the problem of postfix but vaultwarden because it is asking for some matching or may be authentication.
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: connect from unknown[172.17.0.2]
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: smtp_stream_setup: maxtime=300 enable_deadline=0
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 127.0.0.0/8
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 127.0.0.0/8
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? localhost
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? localhost
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 172.17.0.2/32
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 172.17.0.2/32
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 220 bitwarden.mydomain.com ESMTP Postfix
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: watchdog_pat: 0x562f94144700
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: < unknown[172.17.0.2]: EHLO root
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_list_match: unknown: no match
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_list_match: 172.17.0.2: no match
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 250-bitwarden.mydomain.com
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 250-PIPELINING
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 250-SIZE 204800000
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 250-VRFY
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 250-ETRN
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 250-STARTTLS
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 250-ENHANCEDSTATUSCODES
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 250-8BITMIME
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 250-DSN
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 250-SMTPUTF8
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: > unknown[172.17.0.2]: 250 CHUNKING
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: watchdog_pat: 0x562f94144700
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: smtp_get: EOF
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 127.0.0.0/8
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 127.0.0.0/8
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? localhost
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? localhost
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 172.17.0.2/32
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 172.17.0.2/32
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: lost connection after EHLO from unknown[172.17.0.2]
Apr 19 14:56:16 bitwarden postfix/smtpd[279906]: disconnect from unknown[172.17.0.2] ehlo=1 commands=1

**Comment by BlackDex** [comment]
There are no valid authentication methods listed by your SMTP server. That is the issue.
Those are probably not show until you use STARTTLS.

You should see something like:
```
250-AUTH PLAIN LOGIN
```

**Reply by Mickyjohn** [reply]
After setting Force TLS to true the output is,

Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: connect from unknown[172.17.0.2]
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: smtp_stream_setup: maxtime=300 enable_deadline=0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 220 bitwarden.mydomain.com ESMTP Postfix
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]: ????
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad UTF-8 syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]: ??
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad UTF-8 syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]:
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]:
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: smtp_get: EOF
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: lost connection after CONNECT from unknown[172.17.0.2]
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: disconnect from unknown[172.17.0.2] commands=0/0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: connect from unknown[172.17.0.2]
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: smtp_stream_setup: maxtime=300 enable_deadline=0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 220 bitwarden.mydomain.com ESMTP Postfix
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]: ????
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad UTF-8 syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]: ??
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad UTF-8 syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]:
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]:
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: smtp_get: EOF
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: lost connection after CONNECT from unknown[172.17.0.2]
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: disconnect from unknown[172.17.0.2] commands=0/0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: connect from unknown[172.17.0.2]
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: smtp_stream_setup: maxtime=300 enable_deadline=0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 220 bitwarden.mydomain.com ESMTP Postfix
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]: ????
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad UTF-8 syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]: ??
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad UTF-8 syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]:
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]:
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: smtp_get: EOF
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: lost connection after CONNECT from unknown[172.17.0.2]
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: disconnect from unknown[172.17.0.2] commands=0/0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: connect from unknown[172.17.0.2]
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: smtp_stream_setup: maxtime=300 enable_deadline=0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 220 bitwarden.mydomain.com ESMTP Postfix
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]: ????
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad UTF-8 syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]: ??
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad UTF-8 syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]:
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: < unknown[172.17.0.2]:
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: > unknown[172.17.0.2]: 500 5.5.2 Error: bad syntax
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: watchdog_pat: 0x55b2444e7aa0
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: smtp_get: EOF
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 172.17.0.2/32
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: lost connection after CONNECT from unknown[172.17.0.2]
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: disconnect from unknown[172.17.0.2] commands=0/0
[root@bitwarden ldap]#

**Reply by BlackDex** [reply]
I'm not sure which version of Vaultwarden you are using. But you probably do not want to force TLS.
You either need to set `SMTP_SSL=true` or `SMTP_SECURITY=starttls` and maybe you also need 587, but it could work on port 25.
But that will only work if you expose/open port 587 of course when needed.

And besides that, you also need to have correct certs for Postfix.

But, again, this is not the right place to ask for support on Postfix.

**Reply by Masgalor** [reply]
It seems you are a liitle lost when it comes to your postfix configuration.
You should know on what ports postfix awaits submissions and if it expects TLS or not.

You said you tested your setup with
`$ echo "test mail" | sendmail [myname@mydomain.com](mailto:myname@mydomain.com)`
so I am pretty sure your server does not expect TLS.

Something else I have noticed in your logs are those lines
```
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? 127.0.0.0/8
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostaddr: smtpd_client_event_limit_exceptions: 172.17.0.2 ~? localhost
Apr 19 15:19:13 bitwarden postfix/smtpd[285061]: match_hostname: smtpd_client_event_limit_exceptions: unknown ~? 
```
This is your mailserver trying to resolv A and PTR records, this is (depending on your postfix configuration) not necessarry for local submission.

So I guess you problem is something aroung the configuration of:
https://www.postfix.org/postconf.5.html#smtpd_client_restrictions
https://www.postfix.org/postconf.5.html#permit_mynetworks

I am not writing more about those since the misconfiguration of permits would make your server an open relay, so please read the manual and be carefull.

As @BlackDex  already told you, this is defenitly possible with vaultwarden, but you need to understand the configuration of your mailserver.
